protocol/messenger_profile_showcase.go

```diff
@@ -86,3 +124,193 @@ func (m *Messenger) GetProfileShowcasePreferences() (*ProfileShowcasePreferences
+		sharedKey, err := common.MakeECDHSharedKey(m.identity, senderPubKey)
+		if err != nil {
+			return nil, err
+		}
+
+		// Decrypt the main encryption AES key with AES encryption using the DH key
+		dAESKey, err := common.Decrypt(eAESKey, sharedKey)
+		if err != nil {
+			if err.Error() == "cipher: message authentication failed" { // nolint: goconst
+				continue
+			}
+			return nil, err
+		}
+		if dAESKey == nil {
+			return nil, errors.New("decrypting the payload encryption key resulted in no error and a nil key")
+		}
+
+		// Decrypt profile entries with the newly decrypted main encryption AES key
+		entriesData, err := common.Decrypt(encrypted.EncryptedEntries, dAESKey)
+		if err != nil {
+			return nil, err
+		}
+
+		entries := &protobuf.ProfileShowcaseEntries{}
+		err = proto.Unmarshal(entriesData, entries)
+		if err != nil {
+			return nil, err
+		}
+
+		return entries, nil
+	}
+
+	// Return empty if no matching key found
+	return &protobuf.ProfileShowcaseEntries{}, nil
+}
+
+func (m *Messenger) GetProfileShowcaseForSelfIdentity() (*protobuf.ProfileShowcase, error) {
+	preferences, err := m.GetProfileShowcasePreferences()
+	if err != nil {
+		return nil, err
+	}
+
+	forEveryone := &protobuf.ProfileShowcaseEntries{
+		Communities:  toProfileShowcaseUpdateEntries(preferences.Communities, ProfileShowcaseVisibilityEveryone),
+		Accounts:     toProfileShowcaseUpdateEntries(preferences.Accounts, ProfileShowcaseVisibilityEveryone),
+		Collectibles: toProfileShowcaseUpdateEntries(preferences.Collectibles, ProfileShowcaseVisibilityEveryone),
+		Assets:       toProfileShowcaseUpdateEntries(preferences.Assets, ProfileShowcaseVisibilityEveryone),
+	}
+
+	forContacts := &protobuf.ProfileShowcaseEntries{
+		Communities:  toProfileShowcaseUpdateEntries(preferences.Communities, ProfileShowcaseVisibilityContacts),
+		Accounts:     toProfileShowcaseUpdateEntries(preferences.Accounts, ProfileShowcaseVisibilityContacts),
+		Collectibles: toProfileShowcaseUpdateEntries(preferences.Collectibles, ProfileShowcaseVisibilityContacts),
+		Assets:       toProfileShowcaseUpdateEntries(preferences.Assets, ProfileShowcaseVisibilityContacts),
+	}
+
+	forIDVerifiedContacts := &protobuf.ProfileShowcaseEntries{
+		Communities:  toProfileShowcaseUpdateEntries(preferences.Communities, ProfileShowcaseVisibilityIDVerifiedContacts),
+		Accounts:     toProfileShowcaseUpdateEntries(preferences.Accounts, ProfileShowcaseVisibilityIDVerifiedContacts),
+		Collectibles: toProfileShowcaseUpdateEntries(preferences.Collectibles, ProfileShowcaseVisibilityIDVerifiedContacts),
+		Assets:       toProfileShowcaseUpdateEntries(preferences.Assets, ProfileShowcaseVisibilityIDVerifiedContacts),
+	}
+
+	mutualContacts := []*Contact{}
+	iDVerifiedContacts := []*Contact{}
+
+	m.allContacts.Range(func(_ string, contact *Contact) (shouldContinue bool) {
+		if contact.mutual() {
+			mutualContacts = append(mutualContacts, contact)
+			if contact.IsVerified() {
```

Comment:
@MishkaRogachev isn't a verified contact at the same time the mutual contact as well? I guess we should be mutually exclusive here.
```
			if contact.IsVerified() {
							iDVerifiedContacts = append(iDVerifiedContacts, contact)
			} else if contact.mutual() {
							mutualContacts = append(mutualContacts, contact)
			}
```

Reply:
If I understand the identity verification correctly, verifification avaliable only for mutual contacts, so here i explicitly ask for being mutual contacts for identity verified contacts:

| Visibility | For Everyone | For Contacts | For ID Verified Contacts 
| ------ | ------------- | ------------- | ------------- |
Everyone | true | false | false
Mutual contacts | true | true | false
Id verified mutual contacts | true | true | true

Reply:
I agree, but I see that as verified contacts are a subset of mutual contacts, right?

But what I'm pointing you here is that in the current code, the same verified contact, let's say X, will be added to both arrays, `mutualContacts` and `iDVerifiedContacts`. It means that contact X will receive a single message meant to be used for mutual and another message meant to be used for verified contacts, that further means contact X will have a profile showcase for a user that is carried by the last received message, regardless is it mutual to verified contact.

Is that correct?

Reply:
```
		if contact.IsVerified() {
			iDVerifiedContacts = append(iDVerifiedContacts, contact)
		} else if contact.mutual() {
			mutualContacts = append(mutualContacts, contact)
		}
```
With this implementation, a ID verified contact will be able to decrypt `forIDVerifiedContacts` part but will not be able to decrypt `forContacts` part because it won't contain his key. The test will fail [here](https://github.com/status-im/status-go/blob/f1dc7be2aa298323d2b9e29732b2e72666969016/protocol/messenger_profile_showcase_test.go#L359-L360). So i will need to encrypt `forContacts` message both with `iDVerifiedContacts` and `mutualContacts` keys. 

Each 'protobuf.ProfileShowcaseEntries' message contains only those items that are visible to the corresponding category. They are dispatched together in one `protobuf.ProfileShowcase` message, so end user will always use latest  message, regardless is it mutual, verified or neither of them.

Reply:
@saledjenic I guess it's also about the contained data.
A verified contact should process all of the 3 categories to get full showcase. `forIDVerifiedContacts` only contains part of the showcase that's available to verified users.